src/cluster_legacy.c

```diff
@@ -1135,14 +1146,19 @@ void clusterReset(int hard) {
 /* -----------------------------------------------------------------------------
  * CLUSTER communication link
  * -------------------------------------------------------------------------- */
-static clusterMsgSendBlock *createClusterMsgSendBlock(int type, uint32_t msglen) {
-    uint32_t blocklen = msglen + sizeof(clusterMsgSendBlock) - sizeof(clusterMsg);
+static void *createClusterMsgSendBlock(int type, uint32_t msglen) {
+    uint32_t blocklen;
+    if (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT) {
+        blocklen = msglen + sizeof(clusterMsgSendBlockLight) - sizeof(clusterMsgLight);
+    } else {
+        blocklen = msglen + sizeof(clusterMsgSendBlock) - sizeof(clusterMsg);
+    }
     clusterMsgSendBlock *msgblock = zcalloc(blocklen);
     msgblock->refcount = 1;
     msgblock->totlen = blocklen;
     server.stat_cluster_links_memory += blocklen;
     clusterBuildMessageHdr(&msgblock->msg, type, msglen);
-    return msgblock;
+    return (void *)msgblock;
 }
 
 static void clusterMsgSendBlockDecrRefCount(void *node) {
```

Comment:
(Old stuff, so don't feel obliged to change it.) I don't understand why this is `void *`. It looks like just unnecessary casting. It could just be `clusterMsgSendBlock *msgblock` here. That's what it's called with.